src/ex1/script.js

```diff
@@ -0,0 +1,284 @@
+      todo.remove();
+    }, TRASH_ANIMATION_TIMEOUT);
+  }
+}
+
+//clear all button clicked
+function clearAll(e) {
+  const item = e.target;
+  if (item.classList[0] === "clearAllBtn") {
+    todoList.classList.add("fall");
+
+     //delay the remove for animation
+    setTimeout(function () {
+      todoList.innerHTML = "";
+      localStorage.clear();
+      todoList.classList.remove("fall");
+      listStyling(0);
+    }, TRASH_ANIMATION_TIMEOUT);
+    pendingTasksCount.textContent = 0;
+    if (!todoInput.value.length) {
+      inputBox.classList.remove("inactive");
+      todoButton.classList.remove("inactive");
+    }
+    sortBtn.textContent = "Sort";
+  }
+}
+
+//return values from local storage as an array
+function checkLocalStorage() {
+  let todos;
+  if (localStorage.getItem("todos") === null) {
+    todos = [];
+  } else {
+    todos = JSON.parse(localStorage.getItem("todos"));
+  }
+  return todos;
+}
+
+//save data to local storage
+function saveLocalTodos(todo) {
+  let todos = checkLocalStorage();
+  todos.push(todo);
+  localStorage.setItem("todos", JSON.stringify(todos));
+
+  pendingTasksCount.textContent = todos.length;
+}
+
+//get and show the data from local storage after refreshing the page
+function getTodos() {
+  //is the storage empty?
+  let todos = checkLocalStorage();
+
+  todos.forEach(function (todo) {
+    const todoLi = document.createElement("li");
+    todoLi.classList.add("todo");
+    todoLi.innerText = todo;
+
+    //trash button
+    const trashButton = document.createElement("button");
+    trashButton.innerHTML = '<i  class="fas fa-trash"></i>';
+    trashButton.classList.add("trash-btn");
+    todoLi.appendChild(trashButton);
+    //append to list
+    todoList.appendChild(todoLi);
+    //Clear Todo Input value
+    todoInput.value = "";
+  });
+  pendingTasksCount.textContent = todos.length;
+  let todosCount = todoList.childElementCount;
+  listStyling(todosCount);
+
+  if (todos.length) {
+    inputBox.classList.add("inactive");
+    todoButton.classList.add("inactive");
+  }
+}
+
+// remove element from local storage
+function removeLocalTodos(todo) {
+  let todos = checkLocalStorage();
+  const todoIndex = todo.innerText;
+  todos.splice(todos.indexOf(todoIndex), 1);
+  localStorage.setItem("todos", JSON.stringify(todos));
+  pendingTasksCount.textContent = todos.length;
+  listStyling(todos.length);
+  if (!todos.length) {
+    inputBox.classList.remove("inactive");
+    todoButton.classList.remove("inactive");
+    sortBtn.textContent = "Sort";
+  }
+}
+
+//sort function in ascending and descending
+function sortTasks() {
+  let todos = checkLocalStorage();
+  if (!todos.length) {
+    return;
+  }
+  if (todoList.classList.contains("ascending")) {
+    todos.sort(); /* ((a, b) => a.localeCompare(b)) */
+
+    sortBtn.textContent = "Sort 🡱";
+
+    todoList.classList.remove("ascending");
+    todoList.classList.add("descending");
+  } else {
+    todos.sort();
+    todos.reverse();
```

Comment:
Why sort and then reverse? these are 2 consuming array functions.
sort function supports a compareFunc argument - see here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

You can use the sort to both do ascending and descending